docs/DeveloperGuide.md

```diff
@@ -252,51 +252,299 @@ b. Require developers to download those libraries manually (this creates extra w
+Use case ends.  
+
+**Extensions**  
+2a. The list is empty.
+
+> 2a1. Application show an error message.
+   Use case ends.
+
+#### Use case : List all task
+
+**MSS**
+
+1. User request to list all the tasks.
+2. Application show the list of task with respective details.  
+Use case ends.  
+
+**Extensions**  
+2a. The list is empty.
+
+> 2a1. Application show error message.
+   Use case ends.
+   
+#### Use case: Delete task
+
+**MSS**
+
+1. User requests a list of tasks or find task with keyword.
+2. Application shows a list of tasks.
+3. User requests to delete a specific task in the list.
+4. Application request confirmation from the user.
+5. User confirm the confirmation.
+4. Application deletes the task. <br>
+Use case ends.  
+
+**Extensions**  
+
+2a. The list is empty.
+
+> Use case ends.
+
+3a. The given index is invalid.
+
+> 3a1. Application shows an error message.  
+  Use case ends.
+
+#### Use case: edit task
+
+**MSS**
+
+1. User requests a list of tasks or find task with keyword.
+2. Application shows a list of tasks.
+3. User requests to edit a specific task in the list.
+4. Application edit the task. <br>
+Use case ends.  
+
+**Extensions**  
+
+2a. The list is empty.
+
+> Use case ends.
+
+3a. The given index is invalid.
+
+> 3a1. Application shows an error message.  
+  Use case ends.
+
+#### Use case : Undo command
+
+**MSS**
+1. User request to undo command by a specific number.
+2. Application request confirmation from the user.
+3. User confirm the confirmation.
+2. Application undo the command repeatedly based on the given number.
 Use case ends.
 
 **Extensions**
+1a. The given number exceed the total number of tasks.
+> Application will show an error message.
+Use case ends.
+
+#### Use case : Redo command
+
+**MSS**
+1. User request to redo command by a specific number.
+2. Application request confirmation from the user.
+3. User confirm the confirmation.
+4. Application redo the command repeatedly based on the given number.
+Use case ends.
+
+**Extensions**
+1a. The given number exceed the total number of undo commands.
+> Application will show an error message.
+Use case ends.
+
+#### Use case: Complete task
+
+**MSS**
+
+1. User requests a list of tasks or find task with keyword.
+2. Application shows a list of tasks.
+3. User requests to complete a task.
+4. Application complete the task and remove it from the current task list and add into the completed task list.<br>
+Use case ends.  
 
-2a. The list is empty
+**Extensions**  
+
+2a. The list is empty.
+
+> Use case ends.
+
+3a. The given index is invalid.
+
+> 3a1. Application shows an error message.  
+  Use case ends.
+  
+#### Use case: Uncomplete task
+
+**MSS**
+
+1. User requests a list of completed tasks or find compelted task with keyword.
+2. Application shows a list of completed tasks.
+3.  User requests to uncomplete a task.
+4. Application request confirmation from the user.
+5. User confirm the confirmation.
```

Comment:
since we already have support for undo/redo, perhaps removing the confirmation would improve the UX?